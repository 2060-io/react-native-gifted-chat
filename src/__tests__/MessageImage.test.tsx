import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import { MessageImage } from '../GiftedChat'
import { DEFAULT_TEST_MESSAGE } from './data'

// FIXME: Failing due to issue in Lightbox typing
describe('MessageImage', () => {
  it.skip('should not render <MessageImage /> and compare with snapshot', () => {
    const tree = renderer.create(<MessageImage />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it.skip('should  render <MessageImage /> and compare with snapshot', () => {
    const tree = renderer
      .create(
        <MessageImage
          currentMessage={{
            ...DEFAULT_TEST_MESSAGE,
            image: 'url://to/image.png',
          }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
