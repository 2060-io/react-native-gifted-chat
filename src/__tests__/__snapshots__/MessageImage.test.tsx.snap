// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageImage should  render <MessageImage /> and compare with snapshot 1`] = `
<View
  style={
    [
      {},
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      collapsable={false}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Image
          source={
            {
              "uri": "url://to/image.png",
            }
          }
          style={
            [
              {
                "borderRadius": 13,
                "height": 100,
                "margin": 3,
                "resizeMode": "cover",
                "width": 150,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    />
  </View>
</View>
`;

exports[`MessageImage should not render <MessageImage /> and compare with snapshot 1`] = `null`;
